homeassistant/components/agent_dvr/__init__.py

```diff
@@ -0,0 +1,68 @@
+"""Support for Agent."""
+import logging
+
+from agent.a import Agent
+
+from homeassistant.exceptions import ConfigEntryNotReady
+from homeassistant.helpers import device_registry as dr
+
+from .const import DOMAIN as AGENT_DOMAIN, SERVER_URL
+
+ATTRIBUTION = "ispyconnect.com"
+DEFAULT_BRAND = "Agent DVR by ispyconnect.com"
+
+_LOGGER = logging.getLogger(__name__)
+
+DEFAULT_TIMEOUT = 10
+FORWARDS = ["camera"]
+
+
+async def async_setup(hass, config):
+    """Old way to set up integrations."""
+    return True
+
+
+async def async_setup_entry(hass, config_entry):
+    """Set up the Agent component."""
+    hass.data.setdefault(AGENT_DOMAIN, {})
+
+    server_origin = config_entry.data[SERVER_URL]
+
+    agent_client = Agent(server_origin)
+    await agent_client.update()
+    if not agent_client.is_available:
+        raise ConfigEntryNotReady
+
+    agent_client.deviceList = await agent_client.get_devices()
+
+    hass.data[AGENT_DOMAIN][server_origin] = agent_client
```

Comment:
```suggestion
    hass.data[AGENT_DOMAIN][config_entry.entry_id] = agent_client
```

Please use `config_entry.entry_id` as we know its going to be unique.

Reply:
This has broken the integration. I don't want people to be able to add the same server twice which is why I was using server_origin as the ident.

Reply:
When the user is importing via yaml or setting up via the config flow, we should check to make sure the new config entry is going to be unique.  They shouldn't be able to setup the same device twice.